feat(HUB-167): introduce new `rvt-button` for reusability

jog1t opened the pull request:
This partially introduces @NathanFlurry's work from https://github.com/rivet-gg/hub/tree/nathan/new-button.
Changes:
- [x] new `rvt-button`
- [x] adds left slot for using custom elements or icons
- [x] adds right slot for using custom elements or icons
- [x] adds `primary` variant
- [x] adds `secondary` variant
- [x] adds `sm` size
- [x] adds `md` size 
- [x] adds `lg` size
- [x] adds link type
- [x] adds disabled modificator
- [x] adds loading modificator

NathanFlurry commented:
These are awesome. I'm not sure the default should be the 3D button, though. I think the buttons that we have on the landing page are a healthier default:

<img width="222" alt="image" src="https://github.com/rivet-gg/hub/assets/3322446/0aadfc04-4cd8-4600-80e1-79195b28469d">

 Curious on thoughts from @rivet-gg/frontend @NicholasKissel https://feature-hub-167.hub-3av.pages.dev/kitchen-sink

jog1t commented:
@NathanFlurry added a new variant - elevated :)

yusufsallam64 commented:
@jog1t Overall, I really like the look of the new buttons, definitely an upgrade from the old ones, and nice to have things standardized.

Just a few minor nits after poking around a bit,

https://github.com/rivet-gg/hub/assets/68395788/48bf8bd3-98b6-4f69-8fc4-6fa7ac1e5f12

Not sure if this video is the best to show it, but if you click on the button and move off of it, the bottom seems to shift up and kind of throws the animation off. This isn't a huge problem IMO since it's kind of hard to actually do, but would be nice to fix if it's not too difficult.

https://github.com/rivet-gg/hub/assets/68395788/5c847ede-258b-486e-b7c5-a57c703a79aa

Icons seem to shift strangely when you hover over buttons sometimes. I think it's only with the flat buttons; I couldn't seem to reproduce on the extruded ones. Again, this isn't really a big deal, since the icons are small and if you aren't staring at them closely you won't notice this.

Other than that, I think icons in buttons could be moved up a tad since they feel slightly off-center right now. A bit more of a personal suggestion, I think darkening buttons on hover/click or just making them darker when they're disabled would be a nice touch just to provide subtle visual indicators to users.

jog1t commented:
Just added a new form of a bevel effect - I think I might find the one - it uses `:before` and `3d` magic to animate both the button and the bezel at the same time. Performance-wise, it's rendered on GPU, so we can add as many beveled buttons as possible! 😄 Please let me know what you think!

- [x] added a [`tailwind-variants`](https://www.tailwind-variants.org/) library which drastically reduces of mental computation cost when reading what happens in what state :) 
- [x] added missing aria attributes, which increases accessibility 
- [x] added a new animation for standard buttons, a slight scale down when active - the background color slightly dimmers (I'm not convinced to that), let me know what you think!

For now the changes cover only the `primary` variant, will add the `secondary` variant later.

jog1t commented:
Rewritten `secondary` variant to match new design.

NathanFlurry reviewed (changes requested):
See comment above

yusufsallam64 reviewed (changes requested):
Just a few nits:

<img width="221" alt="image" src="https://github.com/rivet-gg/hub/assets/68395788/3e51b3bf-81fd-4dc6-bc92-3d152e830397">

When hovering over these buttons, not a big fan of the different cursor styles since a lot of browsers make it look sort of outdated/cheap. IMO, if a cursor remains default, that already implies the button isn't clickable. 


https://github.com/rivet-gg/hub/assets/68395788/b2c6d3a6-67e1-41e4-bd75-d1366b255ede

I think clicking on buttons makes them resize a _bit_ too much, making it slightly distracting. Reducing this a bit might be nice since I think it should be more of a subtle animation.

<img width="82" alt="image" src="https://github.com/rivet-gg/hub/assets/68395788/71287a20-72a9-4213-87ae-fd148c68ebe8">

Small buttons seem to have a misaligned icon.

Beyond that, maybe an option to give buttons a custom border color (sort of like what @NathanFlurry was talking about in his review)

Other than those things, I like the new buttons a lot! The 3D ones feel especially nice now. Definitely a step up from the old system we had.

jog1t commented:
Current dependencies on/for this PR:
* `main`
  * **PR #23** <a href="https://app.graphite.dev/github/pr/rivet-gg/hub/23?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>  👈
    * **PR #43** <a href="https://app.graphite.dev/github/pr/rivet-gg/hub/43?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/hub/23?utm_source=stack-comment).
